### PR TITLE
Properly handle empty results from test files - fixes #198

### LINKED ISSUE #198: cli: Cannot read property 'stats' of undefined
Hi got this error

```
  √ should load from memory
TypeError: Cannot read property 'stats' of undefined
    at C:\Users\Frontend\AppData\Roaming\nvm\v0.12.5\node_modules\ava\cli.js:139:16
    at Array.map (native)
    at exit (C:\Users\Frontend\AppData\Roaming\nvm\v0.12.5\node_modules\ava\cli.js:138:22)
From previous event:
    at Object.<anonymous> (C:\Users\Frontend\AppData\Roaming\nvm\v0.12.5\node_modules\ava\cli.js:216:18)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```

Here's my code

```js
import test from 'ava';
import { rollup } from 'rollup';

process.chdir(__dirname);

test('should load from memory', t => {
    return rollup({
        entry: '../src/index.js',
        plugins: []
    }).then(bundle => {
        var result = bundle.generate({
            format: 'cjs'
        });

        console.log(result.code);
    });
});
```
When I remove console.log, error does not reject. It doesn't seem problem in rollup.

### PR DESCRIPTION
**Pull request by jamestalmage**
SEE #213 FIRST
Fixes #198.

### CODE REVIEW

File: lib/babel.js [hunk 1]
```diff
@@ -24,11 +24,26 @@ var options = {
...
+};
+
 var transpiled = babel.transformFileSync(testPath, options);
 requireFromString(transpiled.code, testPath, {
 	appendPaths: module.paths
 });
 
+if (!avaRequired) {
+	console.error('no tests found in ' + testPath + ', make sure to import "ava" at the top of your test file');
+	setTimeout(function () {
```
Comment (sindresorhus) [inline lib/babel.js]: `setImmediate`?

File: lib/babel.js [hunk 2]
```diff
@@ -24,11 +24,26 @@ var options = {
...
+	}
+};
+
 var transpiled = babel.transformFileSync(testPath, options);
 requireFromString(transpiled.code, testPath, {
 	appendPaths: module.paths
 });
 
+if (!avaRequired) {
+	console.error('no tests found in ' + testPath + ', make sure to import "ava" at the top of your test file');
```
Comment (sindresorhus) [inline lib/babel.js]:
```
console.error('No tests found in ' + testPath + '. Make sure to import `ava` at the top of your test file.');
```

File: lib/fork.js [hunk 3]
```diff
@@ -34,8 +34,19 @@ module.exports = function (args) {
...
 			if (code > 0 && code !== 143) {
 				reject(new Error(file + ' exited with a non-zero exit code: ' + code));
-			} else {
+			} else if (testResults) {
+				if (!testResults.tests.length) {
+					testResults.stats.failCount++;
+					testResults.tests.push({
+						duration: 0,
+						title: file,
+						error: new Error('no tests for ' + file),
```
Comment (sindresorhus) [inline lib/fork.js]:
```
error: new Error('No tests for ' + file),
```

File: lib/fork.js [hunk 4]
```diff
@@ -34,8 +34,19 @@ module.exports = function (args) {
...
+						type: 'test'
+					});
+				}
 				resolve(testResults);
+			} else {
+				reject(new Error('never got test results from: ' + file));
```
Comment (sindresorhus) [inline lib/fork.js]:
```
reject(new Error('Never got test results from: ' + file));
```

**Comment by jamestalmage** [comment]
I've always been in the custom of putting "fixes..." at the end, but noticed you put it at the beginning in a number of commits. But here you changed it to the end.

Just want to understand the process.

**Comment by sindresorhus** [comment]
I put fixes at the end, and if I squash merge a pull request I put the pull request issue number first.

Squash merge commit for this would be something like:

```
Close PR #208: Properly handle empty results from test files - fixes #198
```

**Comment by jamestalmage** [comment]
gotcha. Fixes are in.

**Comment by sindresorhus** [comment]
:+1:

**Pull request merged**